src/app/[id]/page.tsx

```diff
@@ -0,0 +1,40 @@
+"use client";
+import { useContext, useEffect, useState } from "react";
+// components
+import {
+  CollegeCard,
+  DetailsCards,
+  InstitutionContext,
+} from "@/src/app/components";
+// types
+import { College } from "../types";
+
+const filterCollege = (institutionData: College[], id: number) => {
+  return institutionData.filter((college) => college.id == id)[0];
+};
```

Comment:
Is this used anywhere?

Reply:
yes this gets used on line 25! it filters from the total dataset 
```js
useEffect(() => {
    if (institutionData) {
      setSelectedCollege(filterCollege(institutionData!, params.id));
    }
  }, [institutionData, params.id]);
```

Reply:
Ah, I'm blind. Nice!